lib/pbench/agent/tool_meister.py

```diff
@@ -73,6 +112,245 @@ class ToolException(Exception):
+                self.logger.debug("contents of ansible conf file:%s", temp)
+        except FileNotFoundError:
+            self.logger.exception("ansible conf file does not exist")
+
+        for line in temp.split("\n"):
+            if "roles_path" in line:
+                resp = line
+
+        if resp is not None:
+            self.roles_path = resp.split("=")[-1].strip()
+            self.logger.debug("roles path:%s", self.roles_path)
+        else:
+            self.roles_path = ANSIBLE_ROLES_PATH
+            self.logger.error("not able to find roles path. setting it to default")
+
+    def download_role(self):
+        """ Download role if it does not exist in the localhost """
+        # build command for finding contents of roles path folder
+        command = "ls -lA " + self.roles_path
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        # decode the output
+        output = stdout.decode()
+        self.logger.debug(
+            "output from command:%s is:%s\n and stderr:%s",
+            command,
+            stdout.decode(),
+            stderr,
+        )
+
+        # find if the required role exists or not
+        for line in output:
+            if "performancecopilot.metrics.pcp" in line:
+                # if it exists, return. Dont have to proceed
+                return
+
+        # given role does not exist. We have to download it from ansible galaxy
+        command = "ansible-galaxy collection install performancecopilot.metrics"
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        # decode stderr
+        output_err = stderr
+        self.logger.debug(
+            "output from command:%s is:%s\n and stderr:%s",
+            command,
+            stdout.decode(),
+            stderr,
+        )
+
+        if output_err is not None:
+            self.logger.error("not able to download role. error:%s", output_err)
+
+        # was able to download successfully. set value to true
+        self.downloaded_role = True
+
+    def delete_role(self):
+        """ delete the downloaded role """
+        # create command to remove role
+        command = "ansible-galaxy collection remove performancecopilot.metrics"
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        # decode stderr
+        output_err = stderr
+        self.logger.debug(
+            "output from command:%s is:%s\n and stderr:%s",
+            command,
+            stdout.decode(),
+            stderr,
+        )
+
+        if output_err is not None:
+            self.logger.error("not able to download role. error:%s", output_err)
+
+    def start(self):
```

Comment:
There seems to be a lot of duplication between `start` and `stop` that could probably be factored out into a common method to simplify maintenance.